### ISSUE TITLE
Cargo Input Nodes don't transfer whitelisted Items if not-whitelisted items come first

### ISSUE THREAD

**Issue by BlackBeltPanda** [TheBusyBiscuit/Slimefun4#121]
## Description
With an Input Node in Blacklist mode and no items blacklisted, connected to a chest with an Advanced Output Node with an item whitelisted, if any items are blocking the slots before the whitelisted item in the input chest with nowhere to go, the whitelisted item never gets transported.

## Evidence / Proof
GIF: https://i.gyazo.com/7199e6c7f7e7f173fa3c7f1dfa778efc.gif

## Environment
MC version: 1.11.2 (git-Paper-1032)
CS-CoreLib version: 1.5.12
Slimefun version: 4.1.1

**Comment by ThisTNTSquid** [comment]
Yes, I agree, that somehow causes the sorting system halted when one of the container is fulled

**Comment by BigBadE** [comment]
This is caused by the blacklisted block being added to itemRequests, then being checked later and finding it is blacklisted and ignored.

**Comment by ALiangLiang** [comment]
Is it fixed?

**Comment by BigBadE** [comment]
I personally didn't, it's possible someone else did, but it would require redoing to blacklist system from what I saw.

**Comment by ALiangLiang** [comment]
> This is caused by the blacklisted block being added to itemRequests, then being checked later and finding it is blacklisted and ignored.

I don't think so. Maybe it's just halted by [withdraw](https://github.com/Slimefun/Slimefun4/blob/8536965962e89a57a77c534c3bd3fe8e6d0f031d/src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoNetworkTask.java#L93)?
From my understanding:
1. That code gets the first item of the inventory, and it's in the whitelist or not in the blacklist.
2. The item cannot insert into destination which is fully.
3. Send the item back to the original slot of the inventory.
4. next tick goto 1.

**Comment by BigBadE** [comment]
Keep in mind, I last worked on this 10 months ago, things definitely changed.

**Comment by DarkDaniel107** [comment]
Still experiancing this problem, but with input node with whitelist, and it can't put the items into a Electric Smelter (because the smelter is full).

**Comment by Red-Poppies** [comment]
Just wanted to comment and say that I'm also experiencing this problem when trying to make alloys automatically into an Electric Smeltery. 

Some ingots + dusts may pull into the smeltery but if the smeltery fills up (or if another type of dust/ingot is in front of the one required) it will stop and I end up having to manually put in the 3rd dust or ingot. Hope that makes sense, I am pretty sure it's the same thing shown and what's been mentioned above. 

Having fully automated alloys is my goal atm, so I'll bookmark this and hope for a fix =D! Thanks !

**Comment by Tarluin** [comment]
This also happens on freezer II even with both items whitelisted in advanced output. Your outputting empty buckets and ice, the buckets end up in the first slot, if they don't get removed because the destination is full then it skips moving the ice even though its destination is empty.

**Comment by BigBadE** [comment]
I think I remember what I found originally (again, might of been changed):
1. First item is gotten from the inventory and put into itemRequests
2. Blacklist/whitelist is checked
3. Check fails, item is removed from the list and not transferred
This repeats over and over again because the whitelist/blacklist check only runs on the first item and doesn't fetch a second.

**Comment by Boomer-1** [comment]
Given the age of this and the upcoming cargo rewrite later this year, I would recommend we close this out, and add this to the plans for the cargo rewrite to find a way to check for this

**Comment by qwertyuioplkjhgfd** [comment]
It's still an issue present in the current version, so I would say it's better to keep this documented. If this was closed people may assume it is fixed, and could lead to duplicate issues. Though I agree it may be better to combine this with the other cargo-rewrite-related tasks.
